schema/generate.go

```diff
@@ -791,30 +948,54 @@ func main() {
 		}
 	}
 
+	mapmap := restructureByRemoteObject(schemaFile.Methods)
 	// Generate structs for the remote objects (service APIs)
-	serviceKeys := getSortedServiceKeys(schema.Services)
-	for _, serviceName := range serviceKeys {
-		service := schema.Services[serviceName]
-		if service.Description != "" {
-			fmt.Fprintln(out, formatComment("", service.Description, nil))
-		}
-		printExtensionTags(out, "", service.QlikExtensions)
+	serviceNames := getSortedServiceKeys(mapmap)
+	for _, serviceName := range serviceNames {
+
 		var serviceImplName = serviceName
 		fmt.Fprintln(out, "type ", serviceImplName, "struct {")
 		fmt.Fprintln(out, "\t*RemoteObject")
 		fmt.Fprintln(out, "}")
 
-		methodKeys := getSortedMethodKeys(service.Methods)
+		methodKeys := getSortedMethodKeys(mapmap[serviceName])
 		for _, methodName := range methodKeys {
-			method := service.Methods[methodName]
+			method := mapmap[serviceName][methodName]
+
+			for _, par := range method.Parameters {
+				patchMissingTypeInfo(par.Schema, par.Name, methodName)
```

Comment:
Should this always be called or only when type info is missing? When this info is added to the spec, we don't want any overridden values